### DISCUSSION TITLE
Show constants as tooltip

### DISCUSSION THREAD

**Discussion by Stuartemk** [joomla/joomla-cms#42690] [2. Cold Features]
When debugging the language you should:

1.- Show in the frontend the value of the "normal" text as for all users instead of showing all the constants.
2.- Only when you place the mouse over the text for which you want to discover the constant. Show the constant in the form of a tooltip to be discovered and only to administrators logged in to the frontend.

It is no longer necessary to take the site offline in production since when you enable language debugging from the administration, the value of the text is now shown in the frontend as it is for all users and only the administrators logged in in the frontend are shown the constant in tooltip when placing the mouse over the text (of which you want to discover the constant) that you want to override.

**Comment by brianteeman** [comment]
How do you display a tooltip on a tooltip?

**Comment by ot2sen** [comment]
A different approach could be to have the lang debug part of the system debug plugin follow the settings for 'Allowed groups' and/or the Access setting. Both allow you to control when the main system debug is shown. The language debug tab of the plugin do not have its own setting like those and do not follow those set for system.

**Reply by brianteeman** [reply]
That would be a good change

**Comment by Stuartemk** [comment]
Ideally, when activating the language debugging plugin in the backend, the debugging would be visible in the frontend only for the site administrator, but instead of showing all the constants in the frontend, the value would be shown "normally" but underlined, thanks to that the frontend would not become unreadable and neither would it deform the entire structure of the page as it currently happens, and that when positioning the cursor over the underlined value, the constant would be shown in the tooltip, in this way it is much more readable, more orderly and more intelligent, since generally when doing the debugging it knows exactly what is the value that it wants to overwrite but the constant of that value is unknown. In conclusion, instead of showing all the constants at the same time, it is better to show the values ​​in an identifiable way and only show the constant when hovering over the value. For example: 

`<u>value to change</u>`
